test/test-maintenance/BSCValidatorSet.spec.ts

```diff
@@ -564,4 +564,53 @@ describe('BSCValidatorSet', () => {
     expect(await validatorSet.getMaintainingValidators()).to.deep.eq([]);
     expect(validatorSet.maintainingValidatorSet(0)).to.be.reverted;
   });
+
+  it('common case 1-25: getMiningValidators', async () => {
+    //  set numOfCabinets to 17
+    let govChannelSeq = await crosschain.channelReceiveSequenceMap(GOV_CHANNEL_ID);
+    let govValue = '0x0000000000000000000000000000000000000000000000000000000000000011'; // 17;
+    let govPackageBytes = serializeGovPack('numOfCabinets', govValue, validatorSet.address);
+    await crosschain
+      .connect(operator)
+      .handlePackage(
+        Buffer.concat([buildSyncPackagePrefix(2e16), govPackageBytes]),
+        proof,
+        merkleHeight,
+        govChannelSeq,
+        GOV_CHANNEL_ID
+      );
+    expect(await validatorSet.numOfCabinets()).to.be.eq(BigNumber.from(govValue));
+
+    //  set maxNumOfWorkingCandidates to 9
+    govChannelSeq = await crosschain.channelReceiveSequenceMap(GOV_CHANNEL_ID);
+    govValue = '0x000000000000000000000000000000000000000000000000000000000000000F'; // 15;
+    govPackageBytes = serializeGovPack('maxNumOfWorkingCandidates', govValue, validatorSet.address);
+    await crosschain
+      .connect(operator)
+      .handlePackage(
+        Buffer.concat([buildSyncPackagePrefix(2e16), govPackageBytes]),
+        proof,
+        merkleHeight,
+        govChannelSeq,
+        GOV_CHANNEL_ID
+      );
+    expect(await validatorSet.maxNumOfWorkingCandidates()).to.be.eq(BigNumber.from(govValue));
+
+    var maxNumOfWorkingCandidates = 15;
+    var validators = await validatorSet.getValidators();
+    var epochNumber = 0;
+    var numOfCabinets = 17;
+
+    if ((validators.length - numOfCabinets) < maxNumOfWorkingCandidates){
+      maxNumOfWorkingCandidates = validators.length - numOfCabinets;
+    } 
+    for (var i=epochNumber;i <5;i++){
+      var miningValidators = await validatorSet.getMiningValidators(i);
+      var exceptValues = miningValidators.slice(0,numOfCabinets-maxNumOfWorkingCandidates);
+      for (var j=0;j<numOfCabinets-maxNumOfWorkingCandidates;j++){
```

Comment:
Is not it definitely to be true?

We need to expect that  miningValidators to include `numOfCabinets-maxNumOfWorkingCandidates` canbinates

Reply:
updated